src/sandstorm/grain.capnp

```diff
@@ -336,6 +336,29 @@ interface SandstormApi(AppObjectId) {
   #   failure.
 }
 
+struct DenormalizedGrainMetadata {
+  # The metadata that we need to present contextual information for shared grains (in particular,
+  # information about the app providing that grain, like icon and title).
+
+  appTitle @0 :Util.LocalizedText;
+  # A copy of the app name for the corresponding UIView for presentation in the grain list.
+
+  union {
+    icon :group {
+      format @1 :Text;
+      # Icon asset format, if present.  One of "png" or "svg"
+
+      assetId @2 :Text;
```

Comment:
Maybe it should be 

```capnp
interface StaticAsset {
  getId @0 () -> (assetId :Text);
  getContent @1 () -> (mimeType :Text, data :Data);
}
```

where the `assetId` is the path component of an asset URL. Then these asset ID's would be what we put in `ViewInfo`, and we could add a `staticAssetHost` field to `WebSession.Params`. When an app wants to display an asset, it combines the `staticAssetHost` from the current session with the asset ID.